app/templates/rule_modal.html

```diff
@@ -12,7 +12,8 @@ <h3 class="modal-title" ng-show="!is_edit && is_duplicate">Duplicate Existing Ru
     </div>
     <div class="form-group" ng-class="{'has-error': errors.product}">
       <label for="id_product">Product</label>
-      <input type="text" class="form-control" id="id_product" ng-model="rule.product">
+      <input type="text" class="form-control" id="id_product" ng-model="rule.product"
+       typeahead="product for product in products | filter:$viewValue | limitTo:16">
       <p class="help-block" ng-show="errors.product">{{ errors.product.join(', ') }}</p>
```

Comment:
I think you forgot to add the necessary code here to make channel autocomplete. Product works fine in the UI, but channel does not. This is ready to be merged once that's fixed.